CHANGES.md

```diff
@@ -2,6 +2,8 @@
 
 ## unknown (unknown)
 
+  * Remove all instances of `deriving Typeable`
```

Comment:
It might be worth mentioning that this is because GHC derives this automatically now; we're not actually removing any functionality

Reply:
Do we really need a changelog entry here?

Reply:
My point being: users don't care that we removed some unnecessary statements from our code. It's just noise. The only possibly relevant user-visible change is that we're no longer compatible with GHC < 8.0, but we likely broke that compat earlier elsewhere.

Reply:
Mh that's a good point.  We should just remove this then